### PR TITLE
Change test projects to target netcoreapp2.1

### PR DESCRIPTION
**Pull request by ahsonkhan**
After validating that tests targeting netcoreapp2.0 will pass using the latest packages (https://github.com/dotnet/corefxlab/pull/1735), update the tests to target netcoreapp2.1.

**Do NOT merge until the above mentioned PR is green.**

We could update to 2.1 right away, but we should wait and make sure that the packages we depend on still work on netcoreapp2.0 first.

cc @shiftylogic, @KrzysztofCwalina

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
looks good

**Comment by ahsonkhan** [comment]
FYI, I removed unnecessary or redundant project/package references.

cc @davidfowl

File: src/System.Buffers.Primitives/System.Buffers.Primitives.csproj [hunk 1]
```diff
@@ -8,10 +8,8 @@
     <PackageIconUrl>http://go.microsoft.com/fwlink/?linkid=833199</PackageIconUrl>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="System.Buffers" Version="$(CoreFxVersion)" />
+    <PackageReference Include="System.Buffers" Version="$(CoreFxStableVersion4_4)" />
```
Comment (ahsonkhan) [inline src/System.Buffers.Primitives/System.Buffers.Primitives.csproj]: Question: For a netstandard library, like System.Buffers.Primitives for example, is it better to reference a lower version of a package or higher (4.4.0 vs 4.3.0)? This change results in us referencing the higher version.

cc @ericstj

**Comment by ahsonkhan** [comment]
error CS0433: The type 'ReadOnlySpan' exists in both 'System.Memory, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' [D:\j\workspace\windows_nt_de---4b678204\tests\System.Text.Primitives.Tests\System.Text.Primitives.Tests.csproj]

This issue will be resolved after this PR/fix:
https://github.com/dotnet/corefx/pull/23836

**Comment by ahsonkhan** [comment]
The other issues will get fixed after the cli picks up the new shared framework. Right now it is picking up an old one from Aug 16.

https://github.com/dotnet/cli/pull/7530#discussion_r137400417
https://github.com/dotnet/cli/pull/7601

**Comment by eerhardt** [comment]
FYI - https://github.com/dotnet/cli/pull/7606 moves the CLI to the latest shared framework.

**Comment by ahsonkhan** [comment]
The test failures (such as the one shown below) will go away once we get a new cli with the latest shared framework (which has the AsReadOnlySpan API).

[xUnit.net 00:00:02.2852907]     System.Text.Utf8.Tests.Utf8StringTests.SubstringFromTests(expected: "abbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb", s: "abbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb", substring: "abbbbbbbb") [FAIL]
[xUnit.net 00:00:02.2855085]       System.MissingMethodException : Method not found: 'System.ReadOnlySpan`1<Char> System.Span.AsReadOnlySpan(System.String)'.
[xUnit.net 00:00:02.2856067]       Stack Trace:
[xUnit.net 00:00:02.2856658]            at System.SpanExtensions.AsReadOnlySpan(String text)
[xUnit.net 00:00:02.2857327]         D:\GitHub\Fork\corefxlab\src\System.Text.Utf8String\System\Text\Utf8String.cs(622,0): at System.Text.Utf8.Utf8String.GetUtf8BytesFromString(String str)
[xUnit.net 00:00:02.2858459]         D:\GitHub\Fork\corefxlab\src\System.Text.Utf8String\System\Text\Utf8String.cs(51,0): at System.Text.Utf8.Utf8String..ctor(String s)
[xUnit.net 00:00:02.2859141]         D:\GitHub\Fork\corefxlab\tests\System.Text.Utf8String.Tests\RandomTests.cs(689,0): at System.Text.Utf8.Tests.Utf8StringTests.SubstringFromTests(String expected, String s, String substring)

**Comment by ahsonkhan** [comment]
Waiting for CLI with new shared framework (above 2.1.0-preview1-25719-04) which has the following changes: https://github.com/dotnet/coreclr/pull/14057

**Comment by eerhardt** [comment]
> Waiting for CLI with new shared framework (above 2.1.0-preview1-25719-04) which has the following changes: dotnet/coreclr#14057

@livarcocc @nguerrera @dsplaisted - it looks like the `dotnet/cli` master builds are currently broken.  Is someone fixing them?

**Comment by nguerrera** [comment]
Adding @johnbeisner re: broken master builds. Is it due to signing issues?

File: src/System.Buffers.Primitives/System.Buffers.Primitives.csproj (hunk 1 above)
Comment (ericstj) [reply src/System.Buffers.Primitives/System.Buffers.Primitives.csproj]: Typical best practice is to reference the lowest version that satisfies your dependency (both direct and indirect).  Be careful not to reference a version lower than any of your other dependencies or you will cause a downgrade.

There are sometimes requirements that folks have (limiting test matrix, or limiting number of packages downloaded) that will cause them to break from this best practice and reference higher versions then they actually need, so don't be surprised if you find that this advice is not followed consistently.

**Comment by johnbeisner** [comment]
@nguerrera 
Most likely master is blocked because we took an update to the 'NuGet_Version' from 4.4.0-preview1-4365 to 4.4.0-preview3-4475 [ https://github.com/dotnet/cli/pull/7713 ] which causes the 'Dotnet CLI CI [AllLinuxDistrosNativeInstallers][x64][master]' leg to fail in the 'UploadToLinuxPackageRepository' task at runtime: 
error MSB4018: The "UploadToLinuxPackageRepository" task failed unexpectedly. 
error MSB4018: System.IO.FileLoadException: Could not load file or assembly 'NuGet.Protocol, Version=4.4.0.3, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not find or load a specific file. (Exception from HRESULT: 0x80131621)

https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1030130&_a=summary

**Comment by nguerrera** [comment]
@johnbeisner, you fixed that, right? Is the fix just not RIed to master? Can you do that now?

**Comment by johnbeisner** [comment]
@nguerrera
No I did not fix this - I'm looking into it.

**Comment by ahsonkhan** [comment]
@Vsadov, any idea what could be causing these build failures when we change the test projects to target netcoreapp2.1?

I cannot reproduce this failure locally.

https://ci.dot.net/job/dotnet_corefxlab/job/master/job/windows_nt_release_prtest/819/consoleFull#-75892857179494335-f7bd-47d0-8771-8661e00c2db2
> C:\Users\dotnet-bot\.nuget\packages\microsoft.netcore.compilers\2.6.0-beta1-62126-01\tools\Microsoft.CSharp.Core.targets(84,5): error MSB6006: "RunCsc.cmd" exited with code -2147450749. [D:\j\workspace\windows_nt_re---4abe806d\src\System.Buffers.Primitives\System.Buffers.Primitives.csproj]
17:49:52 C:\Users\dotnet-bot\.nuget\packages\microsoft.netcore.compilers\2.6.0-beta1-62126-01\tools\Microsoft.CSharp.Core.targets(84,5): error MSB6006: "RunCsc.cmd" exited with code -2147450749. [D:\j\workspace\windows_nt_re---4abe806d\src\System.Collections.Sequences\System.Collections.Sequences.csproj]

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
cc @dotnet/corefxlab-collaborators

**Pull request merged**